### PR TITLE
refactor: Clean high-level components from libkb interface

### PR DESCRIPTION
**Pull request by maxtaco**
The goal here is to remove any big objects from the `GenericKey` interface so that in a subsequent PR, we can move `GenericKey` and the structs that implement it into a library that `libkb` depends upon.

### CODE REVIEW

**Comment by maxtaco** [comment]
This PR can be viewed commit by commit, they are each workable checkpoints.  The one CI failure was a flake.

File: go/libkb/generickey.go [hunk 1]
```diff
@@ -12,6 +12,10 @@ import (
 
 type AlgoType int
 
+type VerifyContext interface {
```
Comment (patrickxb) [inline go/libkb/generickey.go]: does anything use this?

Comment (maxtaco) [inline go/libkb/generickey.go]: Yes, `PGPKeyBundle` has a debug output case.

**Comment by patrickxb** [comment]
LGTM.

I couldn't see anywhere where a nil VerifyContext would cause any issues, but I'm not a compiler...I assume there's something somewhere using it.  Otherwise I'd say to get rid of it.

**Comment by maxtaco** [comment]
Thanks for the review.  It's only in the PGP case so we get away with it in the NaCl case we're testing for.

File: go/libkb/pgp_key.go [hunk 2]
```diff
@@ -641,22 +648,22 @@ func (k *PGPKeyBundle) SignToString(msg []byte) (sig string, id keybase1.SigID,
...
-func (k PGPKeyBundle) VerifyStringAndExtract(sig string) (msg []byte, id keybase1.SigID, err error) {
+func (k PGPKeyBundle) VerifyStringAndExtract(ctx VerifyContext, sig string) (msg []byte, id keybase1.SigID, err error) {
 	var ps *ParsedSig
 	if ps, err = PGPOpenSig(sig); err != nil {
 		return
 	} else if err = ps.Verify(k); err != nil {
-		k.G().Log.Debug("Failing key----------\n%s", k.ArmoredPublicKey)
-		k.G().Log.Debug("Failing sig----------\n%s", sig)
+		ctx.Debug("Failing key----------\n%s", k.ArmoredPublicKey)
+		ctx.Debug("Failing sig----------\n%s", sig)
```
Comment (maxtaco) [inline go/libkb/pgp_key.go]: here is where we finally use the `VerifyContext`

Comment (patrickxb) [inline go/libkb/pgp_key.go]: Ok, thanks.

**Pull request merged**